main/exercise/ptest_agree_reorder.class.php

```diff
@@ -0,0 +1,296 @@
+            $buttonGroup[] = $form->addButtonDelete(get_lang('LessAnswer'), 'lessAnswers', true);
+            $buttonGroup[] = $form->addButtonCreate(get_lang('PlusAnswer'), 'moreAnswers', true);
+            $buttonGroup[] = $form->addButton(
+                    'submitQuestion',
+                    $text,
+                    'check',
+                    'primary',
+                    'default',
+                    null,
+                    ['id' => 'submit-question'],
+                    true
+                    );
+            $form->addGroup($buttonGroup);
+        }
+
+        if (!empty($this->id)) {
+            $form->setDefaults($defaults);
+        } else {
+            if ($this->isContent == 1) {
+                // Default sample content.
+                $form->setDefaults($defaults);
+            }
+        }
+        $form->setConstants(['nb_answers' => $nbAnswers]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function processAnswersCreation($form, $exercise)
+    {
+        $objAnswer = new Answer($this->id);
+        $nbAnswers = $form->getSubmitValue('nb_answers');
+
+        for ($i = 1; $i <= $nbAnswers; $i++) {
+            $answer = trim($form->getSubmitValue('answer['.$i.']'));
+            $goodAnswer = false;
+            $comment = '';
+            $weighting = 0;
+            $ptestCategory = (int) $form->getSubmitValue('ptest_category['.$i.']');
+            $dest = '';
+
+            $objAnswer->createAnswer(
+                $answer,
+                $goodAnswer,
+                $comment,
+                $weighting,
+                $i,
+                null,
+                null,
+                $dest,
+                $ptestCategory
+            );
+        }
+
+        // saves the answers into the data base
+        $objAnswer->save();
+
+        $this->save($exercise);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function return_header(Exercise $exercise, $counter = null, $score = [])
+    {
+        $header = parent::return_header($exercise, $counter); //, $score);
+        $header .= '<table class="'.$this->question_table_class.'"><tr>';
+
+        $header .= '<th style="width:1px;white-space:nowrap;">'.get_lang('Choice').'</th>';
+        $header .= '<th>'.get_lang('Answer').'</th>';
+        $header .= '</tr>';
+
+        return $header;
+    }
+
+    /**
+     * Saves one answer to the database.
+     *
+     * @param int    $id          The ID of the answer (has to be calculated for this course)
+     * @param int    $questionId The question ID (to which the answer is attached)
+     * @param string $title       The text of the answer
+     * @param string $comment     The feedback for the answer
+     * @param float  $score       The score you get when picking this answer
+     * @param int    $correct     Whether this answer is considered *the* correct one (this is the unique answer type)
+     */
+    public function addAnswer(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $question_id is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)